cluster/service/cluster.go

```diff
@@ -145,3 +147,39 @@ func (c clusterService) InitializeClusterWatcher() (func() error, error) {
 
 	return watcher.Close, err
 }
+
+// CreateIdentityCluster populate Identity Cluster relationship
+func (c clusterService) CreateIdentityCluster(ctx context.Context, identityID, clusterURL string) error {
+	var err error
+	var id uuid.UUID
+	var rc *repository.Cluster
+
+	id, err = uuid.FromString(identityID)
+	if err != nil {
+		return errors.NewBadParameterError("identity-id", "incorrect Identity ID")
+	}
+
+	rc, err = c.Repositories().Clusters().LoadClusterByURL(ctx, clusterURL)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"cluster_url": clusterURL,
+			"err":         err,
+		}, "failed to load cluster with url %s", clusterURL)
+		return errors.NewBadParameterError("cluster-url", "cluster with requested url doesn't exist")
+	}
+
+	identityCluster := &repository.IdentityCluster{IdentityID: id, ClusterID: rc.ClusterID}
+
+	return c.ExecuteInTransaction(func() error {
+		if err := c.Repositories().IdentityClusters().Create(ctx, identityCluster); err != nil {
+			log.Error(ctx, map[string]interface{}{
```

Comment:
we should not log errors at every stage of the app, IMO. But you could wrap it instead, using `errors.Wrap(...)`